onedal/datatypes/data_conversion.cpp

```diff
@@ -81,7 +81,7 @@ inline dal::homogen_table convert_to_homogen_impl(PyArrayObject *np_data) {
         column_count = static_cast<std::int64_t>(array_size(np_data, 1));
     }
     const auto layout =
-        array_is_behaved_F(np_data) ? dal::data_layout::column_major : dal::data_layout::row_major;
+        array_is_behaved_C(np_data) ? dal::data_layout::row_major : dal::data_layout::column_major;
```

Comment:
Here default layout is changed for the cases where converted numpy array is both C-contagious and F-contagious. It can be happen if the array is of shape (1, n) or (n, 1) (it can be seen in `__setstate__` method here). So it means that from_table/to_table cycle changes layout on oneDAL side. and since currently only C-style onedal tables can be converted to Python (I plan to fix it soon), it prevents multiple pickling. Please see https://github.com/intel/scikit-learn-intelex/pull/1924.